File: client/lib/models/user_model.dart
class User {
  final String id;
  final String activationKey;
  final bool isActive;
  final DateTime activationDate;
  final DateTime? expirationDate;

  User({
    required this.id,
    required this.activationKey,
    required this.isActive,
    required this.activationDate,
    this.expirationDate,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['id'],
      activationKey: json['activationKey'],
      isActive: json['isActive'],
      activationDate: DateTime.parse(json['activationDate']),
      expirationDate: json['expirationDate'] != null
          ? DateTime.parse(json['expirationDate'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'activationKey': activationKey,
      'isActive': isActive,
      'activationDate': activationDate.toIso8601String(),
      'expirationDate': expirationDate?.toIso8601String(),
    };
  }

  bool get isExpired {
    if (expirationDate == null) return false;
    return DateTime.now().isAfter(expirationDate!);
  }
} 

File: server/models/user.model.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const UserSchema = new Schema({
  activationKey: {
    type: String,
    required: true,
    unique: true
  },
  email: {
    type: String,
    default: null
  },
  isActive: {
    type: Boolean,
    default: true
  },
  activationDate: {
    type: Date,
    default: Date.now
  },
  expirationDate: {
    type: Date,
    default: null
  },
  activationLimit: {
    type: Number,
    default: 1,
    min: 1,
    max: 10
  },
  lastLogin: {
    type: Date,
    default: null
  },
  createdBy: {
    type: Schema.Types.Mixed,
    ref: 'Admin',
    required: false,
    default: 'admin-panel'
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

// Обновление поля updatedAt перед сохранением
UserSchema.pre('save', function(next) {
  this.updatedAt = Date.now();
  next();
});

// Метод для проверки активности пользователя
UserSchema.methods.isValidUser = function() {
  if (!this.isActive) return false;
  if (this.expirationDate && new Date() > this.expirationDate) return false;
  return true;
};

module.exports = mongoose.model('User', UserSchema); 